components/script/dom/webidls/ServiceWorkerContainer.webidl

```diff
@@ -6,9 +6,7 @@
 [Pref="dom.serviceworker.enabled", Exposed=(Window,Worker)]
 interface ServiceWorkerContainer : EventTarget {
   [Unforgeable] readonly attribute ServiceWorker? controller;
-  //[SameObject] readonly attribute Promise<ServiceWorkerRegistration> ready;
-
-  [NewObject, Throws] ServiceWorkerRegistration register(USVString scriptURL, optional RegistrationOptions options);
+  [SameObject] readonly attribute Promise<ServiceWorkerRegistration> ready;
```

Comment:
We don't need to implement those methods, they are already implemented. In the `Register` method we just need to call `reject_error` on places where we return an `Err` value and `resolve_native` at places where we return an `Ok` value on the promise object.

You can have a look at [my branch](https://github.com/servo/servo/compare/master...creativcoder:swcontainer_promise?expand=1) here, for how that is being done.